octobot_commons/databases/database_manager.py

```diff
@@ -0,0 +1,136 @@
+        return await self._generate_new_id(back_list=back_list, is_optimizer=True)
+
+    async def _generate_new_id(self, back_list=None, is_optimizer=False):
+        back_list = back_list or []
+        max_runs = constants.MAX_OPTIMIZER_RUNS if is_optimizer else constants.MAX_BACKTESTING_RUNS
+        index = 1
+        while index < max_runs:
+            if index in back_list:
+                index += 1
+                continue
+            name_candidate = self._base_folder(optimizer_id=index) if is_optimizer\
+                else self._base_folder(backtesting_id=index)
+            if self.database_adaptor == adaptors.TinyDBAdaptor:
+                if self._exists(name_candidate):
+                    index += 1
+                else:
+                    return index
+            else:
+                raise RuntimeError(f"Unsupported database adaptor runs ({self.database_adaptor}).")
+        raise RuntimeError(f"Reached maximum number of {'optimizer' if is_optimizer else 'backtesting'} runs "
+                           f"({constants.MAX_BACKTESTING_RUNS}). Please remove some.")
+
+    async def get_optimizer_run_ids(self) -> list:
+        if self.database_adaptor == adaptors.TinyDBAdaptor:
+            optimizer_runs_path = self._merge_parts(self.base_path, constants.OPTIMIZER)
+            if os.path.exists(optimizer_runs_path):
+                return [self._parse_optimizer_id(folder.name)
+                        for folder in os.scandir(optimizer_runs_path)
+                        if os.path.isdir(folder)]
+        return []
+
+    def _parse_optimizer_id(self, identifier) -> str:
+        return identifier.split(constants.DB_SEPARATOR)[-1]
+
+    def _base_folder(self, ignore_backtesting_id=False, backtesting_id=None,
+                     ignore_optimizer_id=False, optimizer_id=None) -> str:
+        path = self.base_path
+        backtesting_id = backtesting_id or self.backtesting_id
+        optimizer_id = optimizer_id or self.optimizer_id
+        if optimizer_id is not None:
+            if ignore_optimizer_id:
+                path = self._merge_parts(path, constants.OPTIMIZER)
+            else:
+                path = self._merge_parts(
+                    path,
+                    constants.OPTIMIZER,
+                    f"{constants.OPTIMIZER}{constants.DB_SEPARATOR}{optimizer_id}"
+                )
+        if backtesting_id is not None:
+            if optimizer_id is None:
+                path = self._merge_parts(path, constants.BACKTESTING)
+            if ignore_backtesting_id:
+                return path
+            return self._merge_parts(path, f"{constants.BACKTESTING}{constants.DB_SEPARATOR}{backtesting_id}")
+        if optimizer_id is None:
+            # live mode
+            return self._merge_parts(path, constants.LIVE)
+        return path
+
+    def _merge_parts(self, *parts):
+        return os.path.join(*parts) \
+            if self.database_adaptor == adaptors.TinyDBAdaptor \
```

Comment:
We'll have to add an `if` for all supported db?

Reply:
Right.
I changed this condition to use `self.database_adaptor.is_file_system_based()` instead